pcdsdevices/sequencer.py

```diff
@@ -0,0 +1,169 @@
+import time
+import logging
+from functools import partial
+
+from ophyd import Device, EpicsSignal, EpicsSignalRO, Component as Cpt
+from ophyd.status import DeviceStatus, SubscriptionStatus
+from ophyd.utils.epics_pvs import raise_if_disconnected
+
+
+logger = logging.getLogger(__name__)
+
+
+class EventSequencer(Device):
+    """
+    Event Sequencer
+
+    The LCLS Event Sequencer implemented as an Flyer; i.e it has the methods
+    :meth:`.kickoff`, :meth:`.complete` and :meth:`.collect`. This allows the
+    EventSequencer to be used succinctly with the `fly_during_wrapper` and
+    associated preprocessor.
+
+    Parameters
+    ----------
+    prefix: str
+        Base prefix of the EventSequencer
+
+    name : str
+        Name of Event Sequencer object
+
+    Examples
+    --------
+    Run the EventSequencer throughout my scan
+
+    .. code::
+
+        fly_during_wrapper(scan([det], motor, ...), [sequencer])
+
+    Run the EventSequencer at each step in my scan, after the completing the
+    motor move.
+
+    .. code::
+
+        def step_then_sequence(detectors, step, pos_cache)
+            yield from one_nd_step(detectors, step, pos_cache)
+            yield from kickoff(sequencer)
```

Comment:
This will move, trigger, read, save, and THEN start the sequencer. Is that really what we want? I think what we really should be doing is move/wait, kickoff, trigger/read/save, complete (despite that being more annoying to write out)

Reply:
I agree. Do you think this level of granularity makes the example more or less confusing?

Reply:
I think, as stated, it doesn't match the description of the example. It'll make the example more confusing if we add granularity. Perhaps we just change the description to make it clear what it does.

Reply:
Seems fine, just want no surprises.